libs/k8s-client/tests/ApiClient/PodsApiClientFunctionalTest.php

```diff
@@ -33,10 +33,47 @@ protected function createResource(array $metadata): Pod
                 'containers' => [
                     [
                         'name' => 'main',
-                        'image' => 'nginx',
+                        'image' => 'alpine',
+                        'command' => ['sh', '-c', 'echo "Hello World"'], // Vypsání a následný spánek
                     ],
                 ],
             ],
         ]);
     }
+
+    public function testReadLogWhenNodeSelectorIsUnsatisfied(): void
+    {
+
+        $pod = $this->createResource([
+            'name' => 'test-resource-1',
+            'labels' => [
+                'app' => 'test-1',
+                self::getTestResourcesLabelName() => (string) getenv('K8S_NAMESPACE'),
+            ],
+        ]);
+
+        $pod->spec->nodeSelector = [
+            'nodePool' => 'dummy-pool',
```

Comment:
simuluje situaci kdy není dostupný žádný node

před updatem clienta, by readLog skončil chybou `Undefined array key 0`

Reply:
Super.. jen bych ten test (nazev) zobecnil na neco jako `testReadLogOfNotStartedPod`. 

Stejne se to bude IMHO chovat, kdyz zkusis cist logy kdykoliv, kdy ten Pod jeste nenajel (ceka na scaleup, ceka na download image, odkazuje na neexistujici configMap atp.)

Reply:
To mi poradilo chat gpt jako nazev testu. Mi prijde ok :D

Reply:
Právě že ne. Takhle specificky se to chová jen když není dostupný node.

Pokud není pullnutej image/nastartovanej pod, tak to skončí exception z toho api.